src/main.ts

```diff
@@ -54,10 +54,17 @@ fetchConfig()
     // start token service to refresh KC token periodically
     console.info('Starting token refresh service...')
     const tokenServices = new TokenServices()
-    tokenServices.initUsingUrl(sessionStorage.getItem('KEYCLOAK_CONFIG_URL') || '')
+    await tokenServices.initUsingUrl(sessionStorage.getItem('KEYCLOAK_CONFIG_URL') || '')
       .then(() => tokenServices.scheduleRefreshTimer())
       .catch((err: string) => console.error(err))
 
+    // Store KeyCloak & Auth roles
+    console.info('Fetching JWT Roles and Authentication roles...')
+    const authService = new AuthenticationService()
+    await authService.getJwtRoles()
+    await authService.getAuthorizations()
```

Comment:
For these, what's the error handling if we can't get the roles/authorizations or they are invalid?

Reply:
Thats a great point, if we wanted to handle the errors in the same way as with Coops-ui ( Dialogs ) it will definitely need a little more work. As you mentioned below this could be solved by doing this after we mount Vue, i will explore a little further though. And yes, 100% we can always do better :)